### PR TITLE
feat: user avatar decorations

### PR DESCRIPTION
**Pull request by aiko-chan-ai**
**Please describe the changes this PR makes and why it should be merged:**

#8914 

> Added:

- `User#avatarDecoration`
- `User#avatarDecorationURL()`

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: src/util/Constants.js [hunk 1]
```diff
@@ -73,6 +73,8 @@ exports.Endpoints = {
         if (dynamic && hash.startsWith('a_')) format = 'gif';
         return makeImageUrl(`${root}/avatars/${userId}/${hash}`, { format, size });
       },
+      AvatarDecoration: (userId, hash, format = 'png', size) =>
```
Comment (Jiralite) [inline src/util/Constants.js:76]: This is missing typings.

Comment (aiko-chan-ai) [reply src/util/Constants.js:76]: oh i forgot it

**Review by kyranet — APPROVED** [review]

File: src/structures/User.js [hunk 2]
```diff
@@ -149,6 +159,16 @@ class User extends Base {
...
 
+  /**
+   * A link to the user's avatar decoration.
+   * @param {StaticImageURLOptions} [options={}] Options for the image URL
+   * @returns {?string}
+   */
+  avatarDecorationURL({ format, size } = {}) {
+    if (!this.avatarDecoration) return null;
+    return this.client.rest.cdn.AvatarDecoration(this.id, this.avatarDecoration, format, size);
+  }
```
[the remark this answers is not in the document]
Comment (aiko-chan-ai) [reply src/structures/User.js:170]: it should look like `avatarURL` .-.

**Review by vladfrangu — APPROVED** [review]

**Comment by Jiralite** [comment]
This pull request needs conflicts to be resolved. Since the branch is now for something else, I'm pretty sure this will need to be remade.

In the future, you should create branches based on what you want to merge into to avoid this problem.

**Comment by aiko-chan-ai** [comment]
Superseded by #9710

**Pull request closed without merging**